waiter/test/waiter/service_description_test.clj

```diff
@@ -206,7 +206,13 @@
                           :expected (str service-id-prefix "fum1234a-a1030ca63357baad681c25935eb4e494")}
                          {:name "service-description->service-id:invalid-chars-present-in--name"
                           :input-data {"name" "fum-!@#$%.,:()"}
-                          :expected (str service-id-prefix "fum-df72716b57632adfc64b74165eb7d7f2")})]
+                          :expected (str service-id-prefix "fum-df72716b57632adfc64b74165eb7d7f2")}
+                         {:name "service-description->service-id:map-data"
+                          :input-data {"env" {"bar" "baz", "fee" "fie", "foe" "fum"}}
+                          :expected (str service-id-prefix "54b332ed6fa91bf23ffc57ef82d3f1ff")}
+                         {:name "service-description->service-id:map-data-reordered"
+                          :input-data {"env" {"foe" "fum", "bar" "baz", "fee" "fie"}}
+                          :expected (str service-id-prefix "54b332ed6fa91bf23ffc57ef82d3f1ff")})]
```

Comment:
These test-cases produce different service-ids without the fix: `test-waiter-prefix-49c060d06a02e18e102568b60abc48a9` and `test-waiter-prefix-cef7f9643e37ba5a5df0469202ddf7e3`, respectively.